src/App.tsx

```diff
@@ -25,7 +25,14 @@ declare module '@mui/styles/defaultTheme' {
 const queryClient = new QueryClient()
 
 function App() {
-  const {id, isLoading, redirect} = useLogin()
+  const {
+    id,
+    isLoadingLoginWithOauth,
+    isLoadingLoginWithUsernameAndPassword,
+    redirect,
+    submitUsernameAndPassword,
+    errorMessageLoginWithUsernameAndPassword,
+  } = useLogin()
```

Comment:
Return additional props from `useLogin` hook, so they can be used in the UsernamePasswordForm via prop drilling. There may be a better way to do this, but leaving that for a future refactor.

Reply:
Can you add a TODO comment (signed and dated) to refactor to a keyed mapping rather than the list of implied "only one" bools? Thanks!

Reply:
Jay noted a similar issue, so I've refactored to pass down an object of the username/password variables, rather than passing down each variable separately. Happy to add a TODO if that doesn't cover the issue noted here!